server/test_utils.go

```diff
@@ -0,0 +1,43 @@
+package main
```

Comment:
I think it makes more sense to call this mock.go as in other files to distinguish it from the test files, there are also `test_utils.go` and `utils_test.go` which is confusing, this way would be less confusing I'd say.